spring-boot-project/spring-boot-autoconfigure/src/test/java/org/springframework/boot/autoconfigure/elasticsearch/ElasticsearchRestClientAutoConfigurationTests.java

```diff
@@ -156,6 +163,69 @@ void restClientCanQueryElasticsearchNode() {
 				});
 	}
 
+	@Test
+	void configureUriWithUsernameOnly() {
+		this.contextRunner.withUserConfiguration(CredentialsProviderConfiguration.class)
+				.withPropertyValues("spring.elasticsearch.rest.uris=http://user@localhost:9200").run((context) -> {
+					RestClient client = context.getBean(RestClient.class);
+					assertThat(client.getNodes().stream().map(Node::getHost).map(HttpHost::toString))
+							.containsExactly("http://localhost:9200");
+
+					CredentialsProvider credentialsProvider = context.getBean(CredentialsProvider.class);
+					Credentials credentials = credentialsProvider.getCredentials(new AuthScope("localhost", 9200));
+					assertThat(credentials.getUserPrincipal().getName()).isEqualTo("user");
+					assertThat(credentials.getPassword()).isNull();
+				});
+	}
+
+	@Test
+	void configureUriWithUsernameAndEmptyPassword() {
+		this.contextRunner.withUserConfiguration(CredentialsProviderConfiguration.class)
+				.withPropertyValues("spring.elasticsearch.rest.uris=http://user:@localhost:9200").run((context) -> {
+					RestClient client = context.getBean(RestClient.class);
+					assertThat(client.getNodes().stream().map(Node::getHost).map(HttpHost::toString))
+							.containsExactly("http://localhost:9200");
+
+					CredentialsProvider credentialsProvider = context.getBean(CredentialsProvider.class);
+					Credentials credentials = credentialsProvider.getCredentials(new AuthScope("localhost", 9200));
+					assertThat(credentials.getUserPrincipal().getName()).isEqualTo("user");
+					assertThat(credentials.getPassword()).isEmpty();
+				});
+	}
+
+	@Test
+	void configureUriWithUsernameAndPasswordWhenUsernameAndPasswordPropertiesSet() {
+		this.contextRunner.withUserConfiguration(CredentialsProviderConfiguration.class)
+				.withPropertyValues("spring.elasticsearch.rest.uris=http://user:password@localhost:9200,localhost:9201",
+						"spring.elasticsearch.rest.username=admin", "spring.elasticsearch.rest.password=admin")
+				.run((context) -> {
+					RestClient client = context.getBean(RestClient.class);
+					assertThat(client.getNodes().stream().map(Node::getHost).map(HttpHost::toString))
+							.containsExactly("http://localhost:9200", "http://localhost:9201");
+
+					CredentialsProvider credentialsProvider = context.getBean(CredentialsProvider.class);
+
+					Credentials uriCredentials = credentialsProvider.getCredentials(new AuthScope("localhost", 9200));
+					assertThat(uriCredentials.getUserPrincipal().getName()).isEqualTo("user");
+					assertThat(uriCredentials.getPassword()).isEqualTo("password");
+
+					Credentials defaultCredentials = credentialsProvider
+							.getCredentials(new AuthScope("localhost", 9201));
+					assertThat(defaultCredentials.getUserPrincipal().getName()).isEqualTo("admin");
+					assertThat(defaultCredentials.getPassword()).isEqualTo("admin");
```

Comment:
Similarly, I think the password in the URI should win so it should be `password`.

Reply:
Sorry, ignore this. I'd missed that there are two URIs being configured, one with credentials and one without.